integration/tests/access/observer_test.go

```diff
@@ -154,7 +167,86 @@ func (s *ObserverSuite) TestObserver() {
 			})
 		}
 	})
+}
 
+// TestObserverRest runs the following tests:
+// 1. CompareRPCs: verifies that the observer client returns the same errors as the access client for rests proxied to the upstream AN
+// 2. HandledByUpstream: stops the upstream AN and verifies that the observer client returns errors for all rests handled by the upstream
+// 3. HandledByObserver: stops the upstream AN and verifies that the observer client handles all other queries
+func (s *ObserverSuite) TestObserverRest() {
+	t := s.T()
+
+	accessAddr := s.net.ContainerByName(testnet.PrimaryAN).Addr(testnet.RESTPort)
+	observerAddr := s.net.ContainerByName("observer_1").Addr(testnet.RESTPort)
+
+	httpClient := http.DefaultClient
+	makeHttpCall := func(method string, url string) (*http.Response, error) {
+		switch method {
+		case http.MethodGet:
+			return httpClient.Get(url)
+		case http.MethodPost:
+			return httpClient.Post(url, "application/json", strings.NewReader("{}"))
+		}
+		panic("not supported")
+	}
+	makeObserverCall := func(method string, path string) (*http.Response, error) {
+		return makeHttpCall(method, "http://"+observerAddr+"/v1"+path)
+	}
+	makeAccessCall := func(method string, path string) (*http.Response, error) {
+		return makeHttpCall(method, "http://"+accessAddr+"/v1"+path)
+	}
+
+	t.Run("CompareEndpoints", func(t *testing.T) {
+		// verify that both clients return the same errors for proxied rests
+		for _, endpoint := range s.getRestEndpoints() {
+			// skip rest handled locally by observer
+			if _, local := s.localRest[endpoint.name]; local {
+				continue
+			}
+			t.Run(endpoint.name, func(t *testing.T) {
+				accessResp, accessErr := makeAccessCall(endpoint.method, endpoint.path)
+				observerResp, observerErr := makeObserverCall(endpoint.method, endpoint.path)
+				assert.NoError(t, accessErr)
+				assert.NoError(t, observerErr)
+				assert.Equal(t, accessResp.Status, observerResp.Status)
+			})
+		}
+	})
+
+	// stop the upstream access container
+	err := s.net.StopContainerByName(context.Background(), testnet.PrimaryAN)
+	require.NoError(t, err)
+
+	t.Run("HandledByUpstream", func(t *testing.T) {
+		// verify that we receive StatusInternalServerError, StatusServiceUnavailable, StatusBadRequest errors from all rests handled upstream
+		for _, endpoint := range s.getRestEndpoints() {
+			if _, local := s.localRest[endpoint.name]; local {
+				continue
+			}
+			t.Run(endpoint.name, func(t *testing.T) {
+				observerResp, observerErr := makeObserverCall(endpoint.method, endpoint.path)
+				require.NoError(t, observerErr)
+				assert.Contains(t, [...]int{
+					http.StatusInternalServerError,
+					http.StatusServiceUnavailable,
+					http.StatusBadRequest}, observerResp.StatusCode)
```

Comment:
is it expected that an endpoint will return `http.StatusBadRequest` when the backend is unavailable? Seems like it should be either an internal error or unavailable.

Reply:
It might be confusing for users if there are 3 different error conditions that could indicate there's a problem with the upstream AN. which endpoints don't return `StatusServiceUnavailable`?